### ISSUE TITLE
Touch-based zooming doesn't work properly in Firefox Nightly or Firefox for Android

### ISSUE THREAD

**Issue by staktrace** [openlayers/openlayers#11614]
**Describe the bug**
When use a two-finger pinch-zoom gesture via a touchscreen in Firefox, zooming sometimes doesn't work quite right and does a double-tap zoom in effect instead.

**To Reproduce**
1. Go to https://openlayers.org/en/latest/examples/zoom-constrained.html in Firefox for Android or in Firefox Nightly on a desktop platform.
2. Put two fingers down **simultaneously** on the screen and pinch-zoom out
3. Observe that instead of pinch-zooming out, a double-tap pinch effect occurs instead.

**Expected behavior**
Instead of a double-tap pinch in, the zoom should follow the gesture done by the fingers.

**Additional context**
I'm a Firefox/Gecko platform developer investigating https://bugzilla.mozilla.org/show_bug.cgi?id=1663731 which was a specific instance of this bug on the lightpollutionmap.info website which uses OpenLayers. However the problem is reproducible even on the hosted OL example sites (e.g. the zoom constrained example which I linked above). This seems like perhaps a case of differing expectations between the events Firefox is sending and what OpenLayers is expecting. If it turns out to be a Firefox bug I'm happy to help get a fix landed on that side.

Note that in order for this bug to reproduce, the `apz.allow_zooming` pref must be enabled in Firefox. This is the default configuration in Firefox for Android and also on desktop Firefox Nightly builds (support on desktop is not on the release channel yet). The pointer events code must also be enabled (which is the case by default in all Firefox release versions). What seems to be happening, per investigation on https://bugzilla.mozilla.org/show_bug.cgi?id=1663731 is that when the second finger goes down, Firefox detects that it can do a pinch-zoom using the two finger gesture, and so dispatches a pointercancel to the content. This somehow triggers the double-tap-zoom-in effect. The reason Firefox detects it can do a pinch-zoom is because there is no touch-action property that disallows it, and because the pref apz.allow_zooming=true enables the pinch zooming codepaths in FIrefox.

The issue can be resolved by setting a `touch-action: none` or `touch-action: pan-x pan-y` property on the ol-viewport element, but from a cursory grep through the changelog it looks like that used to be the case and was removed, so obviously there's some other mechanism that is intended to handle this scenario that may not be working in Firefox.

**Comment by staktrace** [comment]
Reading through the commit at https://github.com/openlayers/openlayers/pull/10332 it seems like the idea there was to call preventDefault() on the touchmove for the second pointer. That's fine for preventing zoom. However the implementation seems to be a bit bound up with the pointer events, and in our case I believe we'll be dispatching the pointercancel before the touchmove, so maybe the implementation's assumptions are being violated.

In other words, in Firefox, if you start a two-finger pinch zoom with simultaneous touches on a zoomable page, the events look something like this:
- pointerdown
- touchstart
- pointerdown
- touchstart
- pointercancel (because there's no touch-action and Firefox determines that a zoom can now occur)
- touchmove

**Comment by staktrace** [comment]
Also in case the Firefox fix doesn't work out, I think an approach that might work here is to call `preventDefault()` on the `touchstart` for the second touch, rather than calling it on the `touchmove`.

**Comment by staktrace** [comment]
The fix on the Firefox side seems to work out ok. I'll close this issue for now and reopen it later if I run into other poblems. Thanks!

**Comment by ahocevar** [comment]
Thanks, @staktrace!
